Source/Core/VideoBackends/D3D12/D3DBase.cpp

```diff
@@ -0,0 +1,859 @@
+// Copyright 2010 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <algorithm>
+
+#include "Common/StringUtil.h"
+#include "VideoBackends/D3D12/D3DBase.h"
+#include "VideoBackends/D3D12/D3DCommandListManager.h"
+#include "VideoBackends/D3D12/D3DDescriptorHeapManager.h"
+#include "VideoBackends/D3D12/D3DState.h"
+#include "VideoBackends/D3D12/D3DTexture.h"
+#include "VideoBackends/D3D12/VertexShaderCache.h"
+#include "VideoCommon/VideoConfig.h"
+
+#define SWAP_CHAIN_BUFFER_COUNT 4
+
+namespace DX12
+{
+
+	HINSTANCE hD3DCompilerDll = nullptr;
+	D3DREFLECT PD3DReflect = nullptr;
+	pD3DCompile PD3DCompile = nullptr;
+	int d3dcompiler_dll_ref = 0;
+
+	CREATEDXGIFACTORY PCreateDXGIFactory = nullptr;
+	HINSTANCE hDXGIDll = nullptr;
+	int dxgi_dll_ref = 0;
+
+	D3D12CREATEDEVICE PD3D12CreateDevice = nullptr;
+	D3D12SERIALIZEROOTSIGNATURE PD3D12SerializeRootSignature = nullptr;
+	D3D12GETDEBUGINTERFACE PD3D12GetDebugInterface = nullptr;
+
+	HINSTANCE hD3DDll12 = nullptr;
+	int d3d12_dll_ref = 0;
+
+	namespace D3D
```

Comment:
Namespaces don't increase indentation in our coding style, all variables should be snake-cased as well (e.g. `this_is_a_variable`)